[Docs] Insert missing Flow variables (#1226)
The-OpenROAD-Project/OpenROAD-flow: docs/user/FlowVariables.md

Note:
-   ☑️ indicates default definition assigned by the tool
-   ✔️ indicates that the variable value may be reassigned with design `config.mk`

> -   ❌indicates that the variable/files is not supported currently.

Reviewer: I think we should keep `=` and `?=` as they correlate with the syntax in `.mk` files. Variables that are not supported should not be documented as there's no benefit.

Author: Some of the variables are supported for a few/ only one platform - should we keep them?

Reviewer: Yes, but in an isolated section. Here we should keep variables that apply to all. We can add them as a subsection under "Optional Variables" at the end of the document.

| `ADDITIONAL_GDS`         | Hardened macro GDS files listed here.                                                              |
| `VERILOG_INCLUDE_DIRS`   | Specifies the include directories for the Verilog input files.                                     |
| `CORNER`                 | PVT corner library selection. Only available for ASAP7 and GF180 PDK.                              |
| `DESIGN_NICKNAME`        | DESIGN_NICKNAME just changes the directory name that ORFS outputs to be DESIGN_NICKNAME instead of DESIGN_NAME in case DESIGN_NAME is unwieldy or conflicts with a difference design.                                                                                    |
| `ABC_AREA`               | Strategies for Yosys ABC synthesis: Area/Speed. Default ABC_SPEED.                                 |
| `PWR_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |
| `GND_NETS_VOLTAGES`      | Used for IR Drop calculation.                                                                      |

> | `BLOCKS`                 | Blocks passed as blackbox into yosys. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile) |

Reviewer 2: ```suggestion
| `BLOCKS`                 | Blocks is a hardened macros through ORFS and integrate to top level design. Do note that you have to specify block-specific inputs file in the directory mentioned by [Makefile](../main/flow/Makefile). |

```

Reviewer 2: @maliberty is description for `BLOCKS` is correct?

Author: @maliberty Any thoughts?

Reviewer 3: Not quite. I would phrase it as this:
`Blocks used as hard macros in a hierarchical flow.`

> | `CDL_FILES`              | Insert additional Circuit Description Language (`.cdl`) netlist files 

Reviewer 2: ```suggestion
| `CDL_FILES`              | Insert additional Circuit Description Language (`.cdl`) netlist files.

```

> | `DFF_LIB_FILES`          | Technology mapping liberty files for flip-flops                                                    |

Reviewer 2: ```suggestion
| `DFF_LIB_FILES`          | Technology mapping liberty files for flip-flops.                                                    |

```

> | `DONT_USE_LIBS`          | Set liberty files as `dont_use`                                                                    |

Reviewer 2: ```suggestion
| `DONT_USE_LIBS`          | Set liberty files as `dont_use`.                                                                    |

```

| `PRESERVE_CELLS`         | Mark modules to keep from getting removed in flattening.                                           |
| `SYNTH_ARGS`             | Optional synthesis variables for yosys.                                                            |

> | `VERILOG_TOP_PARAMS`     | Apply toplevel params (if exist)                                                                   |

Reviewer 2: ```suggestion
| `VERILOG_TOP_PARAMS`     | Apply toplevel params (if exist).                                                                   |

```

| `RESYNTH_AREA_RECOVER` | Enable re-synthesis for area reclaim. |
| `RESYNTH_TIMING_RECOVER` | Enable re-synthesis for timing optimization. | 
| `MACRO_HALO_X`         | Set macro halo for x-direction. Only available for ASAP7 PDK. |
| `MACRO_HALO_Y`         | Set macro halo for y-direction. Only available for ASAP7 PDK. |

## Placement

| Variable                 | Description                                                                                        |
|--------------------------|----------------------------------------------------------------------------------------------------|
| `MACRO_WRAPPERS`         | The wrapper file that replace existing macros with their wrapped version.                          |

## Clock Tree Synthesis

| Variable                 | Description                                                                                        |
|--------------------------|----------------------------------------------------------------------------------------------------|

> | `CTS_BUF_DISTANCE`       | Distance (in microns) between buffers                                                              |

Reviewer 2: ```suggestion
| `CTS_BUF_DISTANCE`       | Distance (in microns) between buffers.                                                              |

```

| `CTS_CLUSTER_DIAMETER`   | Maximum diameter (in microns) of sink cluster. Default 20.                                         |
| `CTS_CLUSTER_SIZE`       | Maximum number of sinks per cluster. Default 50.

## Routing
| Variable                 | Description                                                                                        |
|--------------------------|----------------------------------------------------------------------------------------------------|

> | `FASTROUTE_TCL`          | Specifies a Tcl scripts with commands to run before FastRoute                                      |

Reviewer 2: ```suggestion
| `FASTROUTE_TCL`          | Specifies a Tcl scripts with commands to run before FastRoute.                                      |

```
